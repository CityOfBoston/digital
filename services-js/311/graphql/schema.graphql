type Case {
  id: String!
  service: Service!
  description: String
  status: String!
  statusNotes: String
  serviceNotice: String
  closureReason: String
  closureComment: String
  address: String
  images: [CaseImage!]!
  location: LatLng
  requestedAt: Int
  updatedAt: Int
  expectedAt: Int
  requestedAtString(format: String): String
  updatedAtString(format: String): String
  expectedAtString(format: String): String
  requestedAtRelativeString: String
  updatedAtRelativeString: String
  expectedAtRelativeString: String
}

type CaseImage {
  tags: [String!]!
  squareThumbnailUrl: String!
  squarePreviewUrl: String!
  originalUrl: String!
}

type CaseSearchResults {
  cases: [Case!]!
  query: String!
}

input CreateCaseAttribute {
  code: String!
  value: String!
}

type Geocoder {
  reverse(location: LatLngIn!): Place
  search(query: String!): [Place!]!
}

type LatLng {
  lat: Float!
  lng: Float!
}

input LatLngIn {
  lat: Float!
  lng: Float!
}

enum MetadataRequirement {
  REQUIRED
  VISIBLE
  HIDDEN
}

type Mutation {
  createCase(code: String!, description: String!, descriptionForClassifier: String!, firstName: String, lastName: String, email: String, phone: String, address: String, addressId: String, mediaUrl: String, location: LatLngIn, attributes: [CreateCaseAttribute!]!): Case!
}

type Place {
  location: LatLng!
  address: String!
  addressId: String
  units: [Unit!]!
  exact: Boolean!
}

type Query {
  services: [Service!]!
  topServices(first: Int): [Service!]!
  servicesForDescription(text: String!, max: Int, threshold: Float): [Service!]!
  service(code: String!): Service
  case(id: String!): Case
  searchCases(query: String, topLeft: LatLngIn, bottomRight: LatLngIn): CaseSearchResults!
  geocoder: Geocoder!
}

type Service {
  code: String!
  name: String!
  description: String
  group: String
  attributes: [ServiceAttribute!]!
  locationRequirement: MetadataRequirement!
  contactRequirement: MetadataRequirement!
  locationRequired: Boolean!
  contactRequired: Boolean!
}

type ServiceAttribute {
  type: ServiceAttributeDatatype!
  required: Boolean!
  description: String!
  code: String!
  dependencies: ServiceAttributeConditional
  validations: [ServiceAttributeValidation!]!
  values: [ServiceAttributeValue!]
  conditionalValues: [ServiceAttributeConditionalValues!]
}

type ServiceAttributeCondition {
  attribute: String!
  op: ServiceAttributeConditionalOp!
  value: ServiceAttributeConditionValue!
}

type ServiceAttributeConditional {
  clause: ServiceAttributeConditionalClause!
  conditions: [ServiceAttributeCondition!]!
}

enum ServiceAttributeConditionalClause {
  AND
  OR
}

enum ServiceAttributeConditionalOp {
  eq
  neq
  in
  gt
  gte
  lt
  lte
}

type ServiceAttributeConditionalValues {
  dependentOn: ServiceAttributeConditional!
  values: [ServiceAttributeValue!]!
}

type ServiceAttributeConditionValue {
  type: ServiceAttributeConditionValueType
  string: String
  array: [String!]
  number: Float
}

enum ServiceAttributeConditionValueType {
  STRING
  STRING_ARRAY
  NUMBER
}

enum ServiceAttributeDatatype {
  BOOLEAN_CHECKBOX
  INFORMATIONAL
  MULTIVALUELIST
  NUMBER
  DATETIME
  DATE
  SINGLEVALUELIST
  STRING
  TEXT
}

type ServiceAttributeValidation {
  dependentOn: ServiceAttributeConditional!
  message: String!
  reportOnly: Boolean!
}

type ServiceAttributeValue {
  key: String!
  name: String!
}

type Unit {
  location: LatLng!
  address: String!
  streetAddress: String!
  unit: String!
  addressId: String!
}
