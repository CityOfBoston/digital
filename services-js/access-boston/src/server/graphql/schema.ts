/**
 * @file This file defines the GraphQL schema and resolvers for our server.
 *
 * Run `npm run generate-graphql-schema` to use `ts2gql` to turn this file into
 * the `schema.graphql` file that can be consumed by this and other tools.
 *
 * The output is generated in the “graphql” directory in the package root so
 * that it can be `readFileSync`’d from both `build` (during dev and production)
 * and `src` (during test).
 */
import fs from 'fs';
import path from 'path';

import { makeExecutableSchema } from 'graphql-tools';
import { Resolvers } from '@cityofboston/graphql-typescript';
import AppsRegistry from '../services/AppsRegistry';
import Session from '../Session';
import IdentityIq, { LaunchedWorkflowResponse } from '../services/IdentityIq';
import Boom from 'boom';

/** @graphql schema */
export interface Schema {
  query: Query;
  mutation: Mutation;
}

export interface Query {
  account: Account;
  apps: Apps;
  workflow(args: { caseId: string }): Workflow;
}

export interface Mutation {
  changePassword(args: {
    currentPassword: string;
    newPassword: string;
    confirmPassword: string;
  }): Workflow;

  /**
   * Resets the password. Requires that the user be logged in with the "forgot
   * password" authentication.
   */
  resetPassword(args: {
    newPassword: string;
    confirmPassword: string;
  }): Workflow;
}

export interface Account {
  employeeId: string;
}

export interface Apps {
  categories: AppCategory[];
}

export interface AppCategory {
  title: string;
  showIcons: boolean;
  requestAccessUrl: string | null;

  apps: App[];
}

export interface App {
  title: string;
  url: string;
  iconUrl: string | null;
  description: string;
}

export enum WorkflowStatus {
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
  UNKNOWN = 'UNKNOWN',
}

export enum PasswordError {
  NEW_PASSWORDS_DONT_MATCH = 'NEW_PASSWORDS_DONT_MATCH',
  CURRENT_PASSWORD_WRONG = 'CURRENT_PASSWORD_WRONG',
  NEW_PASSWORD_POLICY_VIOLATION = 'NEW_PASSWORD_POLICY_VIOLATION',
  UNKNOWN_ERROR = 'UNKNOWN_ERROR',
}

export interface Workflow {
  caseId: string | null;
  status: WorkflowStatus;
  messages: string[];
  error: PasswordError | null;
}

// This file is built by the "generate-graphql-schema" script from
// the above interfaces.
const schemaGraphql = fs.readFileSync(
  path.resolve(__dirname, '..', '..', '..', 'graphql', 'schema.graphql'),
  'utf-8'
);

export interface Context {
  session: Session;
  appsRegistry: AppsRegistry;
  identityIq: IdentityIq;
}

const queryRootResolvers: Resolvers<Query, Context> = {
  account: (_root, _args, { session: { loginAuth, forgotPasswordAuth } }) => {
    let employeeId;

    if (loginAuth) {
      employeeId = loginAuth.userId;
    } else if (forgotPasswordAuth) {
      employeeId = forgotPasswordAuth.userId;
    } else {
      throw Boom.forbidden();
    }

    return {
      employeeId,
    };
  },

  apps: (_root, _args, { appsRegistry, session: { loginSession } }) => {
    if (!loginSession) {
      throw Boom.forbidden();
    }

    return {
      categories: appsRegistry
        .appsForGroups(loginSession.groups)
        .map(({ apps, icons, showRequestAccessLink, title }) => ({
          title,
          showIcons: icons,
          requestAccessUrl: showRequestAccessLink ? '#' : null,
          apps: apps.map(({ title, iconUrl, url, description }) => ({
            title,
            iconUrl: iconUrl || null,
            url,
            description,
          })),
        })),
    };
  },

  workflow: async (_root, { caseId }, { identityIq }) =>
    launchedWorkflowResponseToWorkflow(await identityIq.fetchWorkflow(caseId)),
};

const mutationResolvers: Resolvers<Mutation, Context> = {
  changePassword: async (
    _root,
    { currentPassword, newPassword, confirmPassword },
    { identityIq, session: { loginAuth } }
  ) => {
    if (!loginAuth) {
      throw Boom.forbidden();
    }

    if (newPassword !== confirmPassword) {
      return {
        caseId: null,
        error: PasswordError.NEW_PASSWORDS_DONT_MATCH,
        messages: [],
        status: WorkflowStatus.ERROR,
      };
    }

    const workflowResponse = await identityIq.changePassword(
      loginAuth.userId,
      currentPassword,
      newPassword
    );

    return launchedWorkflowResponseToWorkflow(workflowResponse);
  },

  resetPassword: async (
    _root,
    { newPassword, confirmPassword },
    { identityIq, session }
  ) => {
    const { forgotPasswordAuth } = session;

    if (!forgotPasswordAuth) {
      throw Boom.forbidden();
    }

    if (newPassword !== confirmPassword) {
      return {
        caseId: null,
        error: PasswordError.NEW_PASSWORDS_DONT_MATCH,
        messages: [],
        status: WorkflowStatus.ERROR,
      };
    }

    const workflowResponse = await identityIq.resetPassword(
      forgotPasswordAuth.userId,
      newPassword
    );

    if (workflowResponse.completionStatus === 'Success') {
      // If the password was changed successfully, clear the session so they
      // can't reset again.
      //
      // Note that in the non-JS case, this "clear" doesn't actually delete the
      // cookie, since it's called via an "inject" and we don't attempt to
      // propagate Set-Cookie headers back up. But, since the session is stored
      // server-side, we can still invalidate it. The cookie will match to
      // nothing.
      session.reset();
    }

    return launchedWorkflowResponseToWorkflow(workflowResponse);
  },
};

export default makeExecutableSchema({
  typeDefs: [schemaGraphql],
  // We typecheck our own resolvers, so we set this as "any". Otherwise our
  // precise "args" typing conflicts with the general {[argument: string]: any}
  // type that the library gives them.
  resolvers: [
    {
      Query: queryRootResolvers,
      Mutation: mutationResolvers,
    },
  ] as any,
  allowUndefinedInResolve: false,
});

function launchedWorkflowResponseToWorkflow(
  workflowResponse: LaunchedWorkflowResponse
): Workflow {
  let status: WorkflowStatus;

  switch (workflowResponse.completionStatus) {
    case 'Error':
      status = WorkflowStatus.ERROR;
      break;
    case 'Success':
      status = WorkflowStatus.SUCCESS;
      break;
    default:
      status = WorkflowStatus.UNKNOWN;
  }

  const messages = workflowResponse.messages;
  let error: PasswordError | null = null;

  if (status === WorkflowStatus.ERROR) {
    if (messages.find(m => m === 'Current Password Check Failure')) {
      error = PasswordError.CURRENT_PASSWORD_WRONG;
    } else if (
      messages.find(m => m.includes('Password Policy Compliance Failure'))
    ) {
      error = PasswordError.NEW_PASSWORD_POLICY_VIOLATION;
    } else {
      error = PasswordError.UNKNOWN_ERROR;
    }
  }

  return {
    caseId: workflowResponse.id,
    messages,
    error,
    status,
  };
}
