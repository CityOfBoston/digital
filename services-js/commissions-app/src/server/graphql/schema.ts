/**
 * @file This file defines the GraphQL schema and resolvers for our server.
 *
 * Run `npm run generate-graphql-schema` to use `ts2gql` to turn this file into
 * the `schema.graphql` file that can be consumed by this and other tools.
 *
 * The output is generated in the “graphql” directory in the package root so
 * that it can be `readFileSync`’d from both `build` (during dev and production)
 * and `src` (during test).
 */
import fs from 'fs';
import path from 'path';

import { makeExecutableSchema } from 'apollo-server-hapi';
import { Resolvers, ResolvableWith } from '@cityofboston/graphql-typescript';

import CommissionsDao, {
  DbBoard,
  DbDepartment,
  DbMember,
  DbPolicyType,
} from '../dao/CommissionsDao';

/** @graphql schema */
export interface Schema {
  query: Query;
}

export interface Query {
  commissions(args: {
    policyTypeIds: number[] | undefined;
    query: string | undefined;
    hasOpenSeats: boolean | undefined;
  }): Commission[];
  commission(args: { id: number }): Commission | null;

  policyTypes: PolicyType[];
}

export interface PolicyType extends ResolvableWith<DbPolicyType> {
  id: number;
  name: string;
}

export interface Commission extends ResolvableWith<DbBoard> {
  id: number;
  name: string;
  homepageUrl: string | null;
  policyType: PolicyType | null;
  department: Department | null;
  description: string;
  contactName: string;
  contactEmail: string;
  contactPhone: string;
  authority: string | null;
  term: string | null;
  stipend: number;
  seats: number;
  enablingLegislation: string | null;
  members: Member[];
  /**
   * Number of seats that are either vacant or occupied by a holdover whose
   * term is up.
   */
  openSeats: number;
  applyUrl: string;
}

export interface Department extends ResolvableWith<DbDepartment> {
  name: string;
  homepageUrl: string | null;
}

export interface Member extends ResolvableWith<DbMember> {
  name: string;
  appointed: Date | null;
  expires: Date | null;
  status: string;
  holdover: boolean;
}

// This file is built by the "generate-graphql-schema" script from
// the above interfaces.
const schemaGraphql = fs.readFileSync(
  path.resolve(__dirname, '..', '..', '..', 'graphql', 'schema.graphql'),
  'utf-8'
);

export interface Context {
  commissionsDao: CommissionsDao;
}

const caseInsensitiveSearch = (
  query: string,
  text: string | null | undefined
) => text && text.toLocaleLowerCase().includes(query.toLocaleLowerCase());

const queryRootResolvers: Resolvers<Query, Context> = {
  commissions: async (
    _obj,
    { policyTypeIds, query, hasOpenSeats },
    { commissionsDao }
  ) => {
    let commissions = await commissionsDao.fetchBoards();

    // typeof checks because all of the args are optional

    if (Array.isArray(policyTypeIds)) {
      commissions = commissions.filter(
        ({ PolicyTypeId }) =>
          typeof PolicyTypeId === 'number' &&
          policyTypeIds.includes(PolicyTypeId)
      );
    }

    if (typeof query === 'string' && query) {
      commissions = commissions.filter(
        ({ BoardName, Description }) =>
          caseInsensitiveSearch(query, BoardName) ||
          caseInsensitiveSearch(query, Description)
      );
    }

    if (typeof hasOpenSeats === 'boolean') {
      commissions = commissions.filter(
        ({ ActiveCount, Seats }) => (ActiveCount || 0) < Seats === hasOpenSeats
      );
    }

    return commissions;
  },
  commission: (_obj, { id }, { commissionsDao }) =>
    commissionsDao.fetchBoard(id),
  policyTypes: (_obj, _args, { commissionsDao }) =>
    commissionsDao.fetchPolicyTypes(),
};

export const commissionResolvers: Resolvers<Commission, Context> = {
  id: ({ BoardID }) => BoardID,
  name: ({ BoardName }) => BoardName || 'Unknown Board',
  homepageUrl: ({ LinkPath }) => LinkPath,
  description: ({ Description }) => Description || '',
  department: async ({ DepartmentId }, _args, { commissionsDao }) =>
    DepartmentId ? await commissionsDao.fetchDepartment(DepartmentId) : null,
  policyType: async ({ PolicyTypeId }, _args, { commissionsDao }) =>
    PolicyTypeId ? await commissionsDao.fetchPolicyType(PolicyTypeId) : null,
  contactName: ({ Contact }) => Contact || '',
  contactEmail: ({ Email }) => Email || '',
  contactPhone: ({ Phone }) => Phone || '',
  authority: async ({ AuthorityId }, _args, { commissionsDao }) => {
    if (AuthorityId == null) {
      return null;
    }

    const authority = await commissionsDao.fetchAuthority(AuthorityId);

    if (!authority) {
      return null;
    }

    if (!authority.AuthorityType) {
      return null;
    }

    if (authority.AuthorityType.toLocaleLowerCase() === 'not applicable') {
      return null;
    }

    return authority.AuthorityType;
  },
  term: ({ Term }) => Term,
  stipend: ({ Stipend }) => Stipend,
  seats: ({ Seats }) => Seats,
  enablingLegislation: ({ Legislation }) => Legislation,
  // Currently some boards have "0" for the number of seats, so the open seat
  // calculation ends up negative. `max 0` keeps us from looking silly.
  openSeats: ({ ActiveCount, Seats }) =>
    Math.max(0, Seats - (ActiveCount || 0)),
  // TODO(finh): Remove the cityofboston.gov URL when we're stable in
  // production.
  applyUrl: ({ BoardName, BoardID }) =>
    process.env.APPLY_URL
      ? `${process.env.APPLY_URL}?commissionID=${BoardID}`
      : `https://www.cityofboston.gov/boardsandcommissions/application/apply.aspx?bid=${encodeURIComponent(
          BoardName!
        )}`,
  members: async ({ BoardID }, _args, { commissionsDao }) =>
    // Some commissions have employees listed, which we want to remove.
    (await commissionsDao.fetchBoardMembers(BoardID)).filter(
      ({ StatusName }) => StatusName !== 'employee'
    ),
};

const departmentResolvers: Resolvers<Department, Context> = {
  name: ({ DepartmentName }) => DepartmentName || 'Unknown Department',
  homepageUrl: ({ LinkPath }) => LinkPath,
};

const policyTypeResolvers: Resolvers<PolicyType, Context> = {
  id: ({ PolicyTypeId }) => PolicyTypeId,
  name: ({ PolicyType }) => PolicyType,
};

const memberResolvers: Resolvers<Member, Context> = {
  name: ({ FirstName, LastName }) =>
    `${(FirstName || '').trim()} ${(LastName || '').trim()}`.trim(),
  appointed: ({ AppointDtTm }) => AppointDtTm,
  expires: ({ ExpireDtTm }) => ExpireDtTm,
  status: ({ StatusName }) => StatusName,
  holdover: ({ StatusName }) => StatusName === 'Holdover',
};

export default makeExecutableSchema({
  typeDefs: [schemaGraphql],
  // We typecheck our own resolvers, so we set this as "any". Otherwise our
  // precise "args" typing conflicts with the general {[argument: string]: any}
  // type that the library gives them.
  resolvers: [
    {
      Query: queryRootResolvers,
      Commission: commissionResolvers,
      Department: departmentResolvers,
      Member: memberResolvers,
      PolicyType: policyTypeResolvers,
    },
  ] as any,
  allowUndefinedInResolve: false,
});
