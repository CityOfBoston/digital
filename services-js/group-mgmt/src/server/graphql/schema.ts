/**
 * @file This file defines the GraphQL schema and resolvers for our server.
 *
 * Run `npm run generate-graphql-schema` to use `ts2gql` to turn this file into
 * the `schema.graphql` file that can be consumed by this and other tools.
 *
 * The output is generated in the “graphql” directory in the package root so
 * that it can be `readFileSync`’d from both `build` (during dev and production)
 * and `src` (during test).
 */
// import fs from 'fs';
// import path from 'path';
// import { makeExecutableSchema } from 'apollo-server-hapi';
// import { Resolvers } from '@cityofboston/graphql-typescript';

// /** @graphql schema */
// export interface Schema {
//   query: Query;
//   mutation: Mutation;
// }

// export interface Query {};
// export interface Mutation {};

// export interface Context {};

// export type QueryRootResolvers = Resolvers<Query, Context>;
// export type MutationResolvers = Resolvers<Mutation, Context>;
