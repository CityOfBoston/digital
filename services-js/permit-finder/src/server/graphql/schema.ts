/**
 * @file This file defines the GraphQL schema and resolvers for our server.
 *
 * Run `npm run generate-graphql-schema` to use `ts2gql` to turn this file into
 * the `schema.graphql` file that can be consumed by this and other tools.
 *
 * The output is generated in the “graphql” directory in the package root so
 * that it can be `readFileSync`’d from both `build` (during dev and production)
 * and `src` (during test).
 */
import fs from 'fs';
import path from 'path';

import { makeExecutableSchema } from 'apollo-server-hapi';

import { Resolvers, ResolvableWith } from '@cityofboston/graphql-typescript';
import PermitFiles, {
  PermitMilestoneRow,
  PermitReviewRow,
  PermitDbEntry,
} from '../services/PermitFiles';
import { PACKAGE_ROOT } from '../util';

/** @graphql schema */
export interface Schema {
  query: Query;
}

enum PermitKind {
  BUILDING = 'BUILDING',
  FIRE = 'FIRE',
}

interface Query {
  permit(args: { permitNumber: string }): Permit | null;
}

interface Permit extends ResolvableWith<PermitDbEntry> {
  permitNumber: string;
  kind: PermitKind;
  type: string;
  address: string;
  city: string;
  state: string;
  zip: string;

  pointOfContactName: string;

  milestones: Milestone[];
  reviews: Review[];
}

interface Milestone extends ResolvableWith<PermitMilestoneRow> {
  milestoneName: string;

  cityContactName: string | null;

  /** YYYY-MM-DD */
  milestoneStartDate: string;
  /** YYYY-MM-DD */
  milestoneEndDate: string;

  averageDurationSecs: number;
}

interface Review extends ResolvableWith<PermitReviewRow> {
  isAssigned: boolean;
  isStarted: boolean;
  isComplete: boolean;

  status: string;
  type: string;

  reviewerName: string;
}

// This file is built by the "generate-graphql-schema" script from
// the above interfaces.
const schemaGraphql = fs.readFileSync(
  path.resolve(PACKAGE_ROOT, 'graphql', 'schema.graphql'),
  'utf-8'
);

export interface Context {
  permitFiles: PermitFiles;
}

export type QueryRootResolvers = Resolvers<Query, Context>;

const queryRootResolvers: QueryRootResolvers = {
  permit: (_, { permitNumber }, { permitFiles }) =>
    permitFiles.lookupPermit(permitNumber),
};

const permitResolvers: Resolvers<Permit, Context> = {
  address: ({ data: { Address } }) => {
    // The "Address" field also has city and often ZIP, so we pop off everything
    // past the last comma to keep it from showing redundantly.
    const bits = Address.split(',');
    if (bits.length > 1) {
      bits.pop();
    }
    return bits.join(',').trim();
  },
  city: ({ data: { City } }) => City,
  kind: ({ data: { BuildingOrFire } }) =>
    BuildingOrFire === 'Building' ? PermitKind.BUILDING : PermitKind.FIRE,
  permitNumber: ({ data: { PermitNumber } }) => PermitNumber,
  pointOfContactName: ({ data: { PermitPOCName } }) => PermitPOCName,
  state: ({ data: { State } }) => State,
  type: ({ data: { PermitType } }) => PermitType,
  zip: ({ data: { Zip } }) => Zip,

  milestones: ({ milestones }) => milestones,
  reviews: ({ reviews }) => reviews,
};

const milestoneResolvers: Resolvers<Milestone, Context> = {
  milestoneName: ({ MilestoneName }) => MilestoneName,

  cityContactName: ({ CityContactName }) => CityContactName || null,

  // These are in Eastern time, which is how we want to display them, so we can
  // just pull off the date part. Format: 2016-10-03 09:17:21.857000
  milestoneStartDate: ({ MilestoneStartDate }) =>
    MilestoneStartDate.split(' ')[0] || '',
  milestoneEndDate: ({ MilestoneEndDate }) =>
    MilestoneEndDate.split(' ')[0] || '',

  averageDurationSecs: ({ AverageDurationOfMilestone }) =>
    parseInt(AverageDurationOfMilestone, 10),
};

const reviewResolvers: Resolvers<Review, Context> = {
  type: ({ ReviewType }) => ReviewType,
  reviewerName: ({ ReviewerName }) => ReviewerName,
  status: ({ ReviewStatus }) => ReviewStatus,

  isAssigned: ({ IsAssignedFlag }) => IsAssignedFlag === 'Y',
  isStarted: ({ IsStartedFlag }) => IsStartedFlag === 'Y',
  isComplete: ({ IsCompleteFlag }) => IsCompleteFlag === 'Y',
};

export default makeExecutableSchema({
  typeDefs: [schemaGraphql],
  // We typecheck our own resolvers, so we set this as "any". Otherwise our
  // precise "args" typing conflicts with the general {[argument: string]: any}
  // type that the library gives them.
  resolvers: [
    {
      Query: queryRootResolvers,
      Permit: permitResolvers,
      Milestone: milestoneResolvers,
      Review: reviewResolvers,
    },
  ] as any,
  allowUndefinedInResolve: false,
});
