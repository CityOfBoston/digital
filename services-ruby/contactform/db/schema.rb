# This file is auto-generated from the current state of the database. Instead
# of editing this file, please use the migrations feature of Active Record to
# incrementally modify your database, and then regenerate this schema definition.
#
# Note that this schema.rb definition is the authoritative source for your
# database schema. If you need to create the application database on another
# system, you should be using db:schema:load, not running all the migrations
# from scratch. The latter is a flawed and unsustainable approach (the more migrations
# you'll amass, the slower it'll run and the greater likelihood for issues).
#
# It's strongly recommended that you check this file into your version control system.

ActiveRecord::Schema.define(version: 20170607191944) do

  create_table "emails", force: :cascade do |t|
    t.string "name"
    t.string "token"
    t.string "from_address"
    t.string "to_address"
    t.string "subject"
    t.text "message"
    t.string "browser"
    t.string "url"
    t.string "ip"
    t.time "sent"
    t.time "replied"
    t.time "response_time"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end

  create_table "subscribers", force: :cascade do |t|
    t.string "email"
    t.string "zipcode"
    t.string "profile_id"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end

  create_table "users", force: :cascade do |t|
    t.string "name"
    t.string "auth_token"
    t.datetime "created_at", null: false
    t.datetime "updated_at", null: false
  end

end
