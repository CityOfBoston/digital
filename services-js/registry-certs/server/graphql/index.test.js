// @flow
// graphql-tools/schema
import schema from './index.ts';

describe('schema', () => {
  it('is generated', () => {
    expect(schema).toMatchSnapshot();
  });
});
