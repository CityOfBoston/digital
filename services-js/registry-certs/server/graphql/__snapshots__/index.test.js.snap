// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema is generated 1`] = `
GraphQLSchema {
  "__allowedLegacyNames": Array [],
  "__validationErrors": undefined,
  "_directives": Array [
    "@skip",
    "@include",
    "@deprecated",
  ],
  "_implementations": Object {},
  "_mutationType": "Mutation",
  "_possibleTypeMap": Object {},
  "_queryType": "Query",
  "_subscriptionType": null,
  "_typeMap": Object {
    "BirthCertificateOrderItemInput": "BirthCertificateOrderItemInput",
    "Boolean": "Boolean",
    "CertificateOrderItem": "CertificateOrderItem",
    "ChargeOrderError": "ChargeOrderError",
    "ChargeOrderErrorCode": "ChargeOrderErrorCode",
    "ChargeOrderResult": "ChargeOrderResult",
    "Date": "Date",
    "DeathCertificate": "DeathCertificate",
    "DeathCertificateOrder": "DeathCertificateOrder",
    "DeathCertificateOrderItemInput": "DeathCertificateOrderItemInput",
    "DeathCertificateSearch": "DeathCertificateSearch",
    "DeathCertificates": "DeathCertificates",
    "DeleteUploadResult": "DeleteUploadResult",
    "Float": "Float",
    "Int": "Int",
    "MarriageCertificateOrderItemInput": "MarriageCertificateOrderItemInput",
    "Mutation": "Mutation",
    "OrderError": "OrderError",
    "OrderErrorCause": "OrderErrorCause",
    "OrderResult": "OrderResult",
    "OrderType": "OrderType",
    "Query": "Query",
    "String": "String",
    "SubmittedOrder": "SubmittedOrder",
    "__Directive": "__Directive",
    "__DirectiveLocation": "__DirectiveLocation",
    "__EnumValue": "__EnumValue",
    "__Field": "__Field",
    "__InputValue": "__InputValue",
    "__Schema": "__Schema",
    "__Type": "__Type",
    "__TypeKind": "__TypeKind",
  },
  "astNode": Object {
    "directives": Array [],
    "kind": "SchemaDefinition",
    "loc": Object {
      "end": 4702,
      "start": 4656,
    },
    "operationTypes": Array [
      Object {
        "kind": "OperationTypeDefinition",
        "loc": Object {
          "end": 4685,
          "start": 4667,
        },
        "operation": "mutation",
        "type": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 4685,
            "start": 4677,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 4685,
              "start": 4677,
            },
            "value": "Mutation",
          },
        },
      },
      Object {
        "kind": "OperationTypeDefinition",
        "loc": Object {
          "end": 4700,
          "start": 4688,
        },
        "operation": "query",
        "type": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 4700,
            "start": 4695,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 4700,
              "start": 4695,
            },
            "value": "Query",
          },
        },
      },
    ],
  },
  "extensionASTNodes": undefined,
}
`;
